integration/presto/src/main/java/org/apache/carbondata/presto/CarbonColumnVectorWrapper.java

```diff
@@ -273,7 +279,7 @@ public boolean hasDictionary() {
 
   @Override
   public CarbonColumnVector getDictionaryVector() {
-    return this.columnVector.getDictionaryVector();
+    return this.dictionaryVectorWrapper;
```

Comment:
Currently in our code it is used only in `SliceStreamReader#buildBlock`.
I don't see any problem here.

Also in your issue JIRA,  saw that it is in delete scenario, **our version of carbon doesn't support insert delete from presto yet**. Feel free to contribute.

Reply:
Thanks for the review.
JIRA issue I created had insert and delete carried out using spark integration; and read using presto integration.

I'm not sure about the applicability part, I trust the contributes and maintainers wisdom in here :)
However, I observed the case while my local verification and suggested the fix after verification.
Feel free to use or reject.

Reply:
I reviewed the `SliceStreamReader::buildBlock`, I dont see how it can filter out the deleted record when `carbon.push.rowfilters.for.vector=true` property is set.

Reply:
ok, thanks for the reply. I got confused with presto delete update. 
Please add a test case that can reproduce issue in the suite.